internal/rbd/types/mirror.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2024 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import (
+	"context"
+
+	"github.com/ceph/ceph-csi/internal/util"
+
+	librbd "github.com/ceph/go-ceph/rbd"
+	"github.com/ceph/go-ceph/rbd/admin"
+)
+
+type Mirror interface {
+	// EnableMirroring enables mirroring on the resource with the specified mode.
+	EnableMirroring(mode librbd.ImageMirrorMode) error
+	// DisableMirroring disables mirroring on the resource with the option to force the operation
+	DisableMirroring(force bool) error
+	// Promote promotes the resource to primary status with the option to force the operation
+	Promote(force bool) error
+	// ForcePromote promotes the resource to primary status with a timeout
+	ForcePromote(cr *util.Credentials) error
+	// Demote demotes the resource to secondary status
+	Demote() error
+	// Resync resynchronizes the resource
+	Resync() error
+
+	// GetMirroringInfo returns the mirroring information of the resource
+	GetMirroringInfo() (interface{}, error)
```

Comment:
It is really ugly to return an untyped interface. Can that be improved?

Reply:
i tried but it gets more difficult for implementation where i need to have different duplicate checks for both volume and group, that's the  reason to settle on this after lot of other considerations

Reply:
Maybe add a `MirrorInfo` interface too?

Reply:
Am not sure adding MirrorInfo interface helps but the problem is that the volumemirror and groupmirror in go-ceph return different structs, if both are returning the same i could have solved this or else I need to own a local struct and fill the required values from both, i can do this if you prefer.

Reply:
Maybe, or have a MirrorInfo interface with only some basic function, or completely empty? At least the type can then describe that it is actually one of the go-ceph types.